x-pack/plugins/cases/server/authorization/audit_logger.test.ts

```diff
@@ -0,0 +1,208 @@
+      const authLogger = new AuthorizationAuditLogger();
+      jest.spyOn(authLogger, 'log');
+
+      expect(() => {
+        authLogger.log({
+          operation: Operations.createCase,
+          entity: {
+            owner: 'a',
+            id: '1',
+          },
+        });
+      }).not.toThrow();
+
+      expect(authLogger.log).toHaveBeenCalledTimes(1);
+    });
+
+    it('logs a message with a saved object ID in the message field', () => {
+      logger.log({
+        operation: Operations.createCase,
+        entity: {
+          owner: 'a',
+          id: '1',
+        },
+      });
+      expect(mockLogger.log.mock.calls[0][0]?.message).toContain('[id=1]');
+    });
+
+    it('creates the owner part of the message when no owners are specified', () => {
+      logger.log({
+        operation: Operations.createCase,
+      });
+
+      expect(mockLogger.log.mock.calls[0][0]?.message).toContain('as any owners');
+    });
+
+    it('creates the owner part of the message when an owner is specified', () => {
+      logger.log({
+        operation: Operations.createCase,
+        entity: {
+          owner: 'a',
+          id: '1',
+        },
+      });
+
+      expect(mockLogger.log.mock.calls[0][0]?.message).toContain('as owner "a"');
+    });
+
+    it('creates a failure message when passed an error', () => {
+      logger.log({
+        operation: Operations.createCase,
+        entity: {
+          owner: 'a',
+          id: '1',
+        },
+        error: new Error('error occurred'),
+      });
+
+      expect(mockLogger.log.mock.calls[0][0]?.message).toBe(
+        'Failed attempt to create cases [id=1] as owner "a"'
+      );
+
+      expect(mockLogger.log.mock.calls[0][0]?.event?.outcome).toBe('failure');
+    });
+
+    it('creates a write operation message', () => {
+      logger.log({
+        operation: Operations.createCase,
+        entity: {
+          owner: 'a',
+          id: '1',
+        },
+      });
+
+      expect(mockLogger.log.mock.calls[0][0]?.message).toBe(
+        'User is creating cases [id=1] as owner "a"'
+      );
+
+      expect(mockLogger.log.mock.calls[0][0]?.event?.outcome).toBe('unknown');
+    });
+
+    it('creates a read operation message', () => {
+      logger.log({
+        operation: Operations.getCase,
+        entity: {
+          owner: 'a',
+          id: '1',
+        },
+      });
+
+      expect(mockLogger.log.mock.calls[0][0]?.message).toBe(
+        'User has accessed cases [id=1] as owner "a"'
+      );
+
+      expect(mockLogger.log.mock.calls[0][0]?.event?.outcome).toBe('success');
+    });
+
+    describe('event structure', () => {
+      // I would have preferred to do these as match inline but that isn't supported because this is essentially a for loop
+      // for reference: https://github.com/facebook/jest/issues/9409#issuecomment-629272237
+
+      // This loops through all operation keys
+      it.each(Array.from(Object.keys(Operations)))(
```

Comment:
Do you need `Array.from` here?  `Object.keys` returns an array

Reply:
Yeah good point, I think I had Object.entries at one point which returned an iterator and forgot to remove it 👍

Reply:
https://github.com/elastic/kibana/pull/101325/commits/eadbfd8501992c7736703270e09731dfd950b105